### DISCUSSION TITLE
Get system current directory from MediaFileStore

### DISCUSSION THREAD

**Discussion by dor1202** [OrchardCMS/OrchardCore#12832] [Q&A] [answered]
Hi.
I'm using `IMediaFileStore` interface, how can I get a directory URL?
I mean, if I search for `await _mediaFileStore.CreateFileFromStreamAsync("/test");`
I will search in the media folder for the wanted path and return  for example `C:\Software\sample\App_Data\Sites\Default\Media\test`

**Comment by jtkech** [comment, marked as answer]
`IMediaFileStore` inherits from `IFileStore` that exposes a `GetFileInfoAsync()` method.

**Reply by dor1202** [reply]
Hi.
Thank you for the response, I'm adding an example code for you to see:
```cs
var tmp = await _mediaFileStore.GetFileInfoAsync("test/sample.zip");
var a = tmp.Path; // output: test/sample.zip
var b = tmp.DirectoryPath; // output: test
```
They aren't returning the full path.

**Reply by jtkech** [reply]
Ah okay, look at what we do in the `Startup.cs` file of the Media module.
  
    // The following can be injected from the DI in a class constructor.
    var shellOptions = serviceProvider.GetRequiredService<IOptions<ShellOptions>>();
    var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
    var options = serviceProvider.GetRequiredService<IOptions<MediaOptions>>().Value;
    ...
    var mediaPath = GetMediaPath(shellOptions.Value, shellSettings, options.AssetsPath);
    ....

    private string GetMediaPath(ShellOptions shellOptions, ShellSettings shellSettings, string assetsPath)
    {
        return PathExtensions.Combine(
            shellOptions.ShellsApplicationDataPath,
            shellOptions.ShellsContainerName,
            shellSettings.Name, assetsPath);
    }

**Reply by dor1202** [reply]
@jtkech, thank you it worked flawlessly.
